Source/BugsnagClient.m

```diff
@@ -843,6 +873,68 @@ - (void)updateCrashDetectionSettings {
         [self.oomWatchdog disable];
     }
 }
+
+/**
+ * Configure event listeners for automatic breadcrumbs.
+ */
+- (void)updateAutomaticBreadcrumbDetectionSettings {
+
+    // No automatic breadcrumbs are enabled; remove notifications for them all and return.
+    if (![[self configuration] enabledBreadcrumbTypes]) {
+        NSArray *allEventNames = [[[[self automaticBreadcrumbStateEvents]
+            arrayByAddingObjectsFromArray:[self automaticBreadcrumbControlEvents]]
+            arrayByAddingObjectsFromArray:[self automaticBreadcrumbMenuItemEvents]]
+            arrayByAddingObjectsFromArray:[self automaticBreadcrumbTableItemEvents]];
+        
+        for (NSString *eventName in allEventNames) {
+            [[NSNotificationCenter defaultCenter] removeObserver:self
+                                                            name:eventName
+                                                          object:nil];
+        }
+        return;
+    }
```

Comment:
I don't believe we don't need to remove observers as this is called on `start`, so they won't exist already.

Reply:
Yup, legacy code moved around without due consideration.